lib/data/repository/genre_repository.dart

```diff
@@ -1,16 +1,17 @@
 import 'dart:convert';
 
 import 'package:flutter/services.dart';
-import '../entities/genre.dart';
+import '../../domain/repository/genre_repository_interface.dart';
+import '../../domain/entity/genre.dart';
 
-class GenreRepository {
+class GenreRepository implements GenreRepositoryInterface {
```

Comment:
Minor: Personally I'd choose one of the below naming conventions:

- `class GenreRepositoryImpl implements GenreRepository`
- `class GenreRepository implements IGenreRepository`
- `class LocalGenreRepository implements GenreRepository`

Thus avoiding the `Interface` ending. Personal choice! If you don't agree with this, mark this as resolved by yourself :)

Reply:
seen second approach quite often... so changed to  class GenreRepositoryImp implements IGenRepository
Thanks!